File: Chapter2/2.3/2.3.2-Primary Arithmetic/2.3.2-Primary Arithmetic.cpp
#include <iostream>
using namespace std;

int main(){
    int a,b;
    while (cin>>a>>b){
        if (a==0 && b==0){
            break;
        }
        else{
            int c=0,count=0;
            for (int i=9;i>=0;i--){
                c = (a%10+b%10+c)>9?1:0;
                count+=c;
                a/=10;
                b/=10;
            }
        if (count==0){
            cout<<"No carry operation."<<endl;
        }
        else if (count==1){
            cout<<"1 carry operation."<<endl;
        }
        else{
            cout<<count<<" carry operations."<<endl; 
        }
        }
    }
}

File: Chapter2/2.3/2.3.2-Primary Arithmetic/2.3.2-Primary Arithmetic.py
#Runtime ERROR
while True:
    a,b=map(str,input().split())
    if a=='0' and b=='0':
        break
    else:
        add_=0
        count=0
        for i in range(len(a)-1,-1,-1):
            num=int(a[i])+int(b[i])+add_
            while num>=10:
                count+=1
                add_+=1
                num-=10
    if count==0:
        print('No carry operation.')
    elif count==1:
        print(f'{count} carry operation.')
    else:
        print(f'{count} carry operations.')
    